js/comm.js

```diff
@@ -15,7 +15,8 @@
 Haskell.createWebSocket = function (url0, receive) {
   var that = {};
   var optReloadOnDisconnect = false;
-  var url  = 'ws:' + url0.slice(5) + '/websocket/';
+  var connectType = location.protocol === 'https:' ? 'wss' : 'ws';
```

Comment:
Aren't the colons missing from the protocol prefixes?

```suggestion
  var connectType = location.protocol === 'https:' ? 'wss:' : 'ws:';
```

Reply:
The slice was including the ':' originally, so there were two colons in the strings 'ws::'.  @HeinrichApfelmus  fixed this with his update.